plugins/search/src/components/SearchModal/SearchModal.tsx

```diff
@@ -0,0 +1,140 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import {
+  Paper,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  List,
+  Grid,
+} from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import { SearchBarBase } from '../SearchBar';
+import { DefaultResultListItem } from '../DefaultResultListItem';
+import { SearchResult } from '../SearchResult';
+import { SearchContextProvider, useSearch } from '../SearchContext';
+import { SearchResultPager } from '../SearchResultPager';
+
+import { useDebounce } from 'react-use';
+
+interface SearchModalProps {
+  open?: boolean;
+  toggleModal: () => void;
+}
+
+const useStyles = makeStyles(() => ({
+  container: {
+    borderRadius: 30,
+    display: 'flex',
+    height: '2.4em',
+  },
+  input: {
+    flex: 1,
+  },
+  paperFullWidth: { minHeight: 'calc(100% - 128px)' },
+  dialogActionsContainer: { padding: '8px 24px' },
+}));
+
+export const Modal = ({ open = true, toggleModal }: SearchModalProps) => {
+  const classes = useStyles();
+
+  const { term, setTerm } = useSearch();
+  const [value, setValue] = useState<string>(term);
+
+  useEffect(() => {
+    setValue(prevValue => (prevValue !== term ? term : prevValue));
+  }, [term]);
+
+  useDebounce(() => setTerm(value), 1000, [value]);
+
+  const handleQuery = (newValue: string) => {
+    setValue(newValue);
+  };
+
+  const handleClear = () => setValue('');
+
+  const handleResultClick = () => {
+    toggleModal();
+    handleClear();
+  };
+
+  const handleKeyPress = () => {
+    handleResultClick();
+  };
+
+  return (
+    <Dialog
+      classes={{
+        paperFullWidth: classes.paperFullWidth,
+      }}
+      onClose={toggleModal}
+      aria-labelledby="customized-dialog-title"
+      open={open}
+      fullWidth
+      maxWidth="lg"
+    >
+      <DialogTitle id="customized-dialog-title">
+        <Paper className={classes.container}>
+          <SearchBarBase
+            className={classes.input}
+            value={value}
+            onChange={handleQuery}
+            onClear={handleClear}
+          />
+        </Paper>
+      </DialogTitle>
+      <DialogContent>
+        <SearchResult>
+          {({ results }) => (
+            <List>
+              {results.map(({ document }) => (
+                <div
+                  role="button"
+                  tabIndex={0}
+                  key={`${document.location}-btn`}
+                  onClick={handleResultClick}
+                  onKeyPress={handleKeyPress}
+                >
+                  <DefaultResultListItem
```

Comment:
This has me wondering how we might be able to enable App Integrators to customize/compose this.  What if I as an Integrator want it to use the TechDocs result list item for TechDocs results?  Or the Catalog result list item for Catalog results?

Not something we need to address in this PR, but something to think about for the future.